src/components/tag/Tag.tsx

```diff
@@ -1,58 +1,116 @@
+  float: left;
   align-items: center;
-  text-transform: capitalize;
-  text-align: center;
-  font-family: ${theme.fonts.body};
-  font-size: ${theme.fontSizes.sm};
-  line-height: ${theme.lineHeight};
   padding: 0 ${theme.spacing.spacing03};
-  margin: ${theme.spacing.spacing01};
-  border-radius: 1rem;
-  color: ${theme.colors.lights[0]};
-  background-color: ${({ color }) =>
-    color ? theme.colors[color] || color : theme.colors.lights[3]};
-  cursor: ${({ onClick }) => onClick && "pointer"};
-  span {
-    width: 0.8rem;
-    height: 0.8rem;
-    display: block;
-    font-size: 0.5rem;
-    font-weight: bold;
-    margin-left: 0.4rem;
-    margin-bottom: 0.2rem;
-    transform: translate(2px, 1px);
+  font-size: ${theme.fontSizes.sm};
+  line-height: 1.5rem;
+  background: ${theme.colors.lights[2]};
+  border-radius: ${theme.radius.xxl};
+  margin: 0 4px 4px 0;
+
+  .tag__icon,
+  .tag__close-button {
+    display: flex;
+    align-items: center;
+    justify-content: center;
+    width: 1rem;
+    height: 1rem;
+  }
+
+  .tag__icon {
+    margin: 0 0.25rem 0 -0.375rem;
+  }
+
+  .tag__close-button {
+    border-radius: ${theme.radius.xxl};
+    background: ${darken(0.1, theme.colors.lights[2])};
+    margin: 0 -0.375rem 0 0.5rem;
+
+    svg {
+      width: 0.625rem;
+      height: 0.625rem;
+    }
   }
+
   &:hover {
-    opacity: ${({ onClick }) => onClick && "0.8"};
+    background: ${darken(0.05, theme.colors.lights[2])};
+    cursor: pointer;
+    transition: all 0.2s ease-in-out;
+    .tag__text {
+      transition: all 0.2s ease-in-out;
+      opacity: 0.8;
+    }
+  }
+  &:active {
+    background: ${darken(0.1, theme.colors.lights[2])};
+    cursor: pointer;
   }
+
+  ${({ color }) =>
+    color &&
+    css`
+      background: ${theme.colors[color] || color};
+      color: ${theme.colors.snowWhite};
+      svg {
+        fill: ${theme.colors.snowWhite};
+      }
+      .tag__close-button {
+        background: ${darken(0.1, theme.colors[color] || color)};
+      }
+
+      &:hover {
+        background: ${darken(0.05, theme.colors[color] || color)};
+      }
+      &:active {
+        background: ${darken(0.1, theme.colors[color] || color)};
+      }
+    `}
+  ${({ colorText }) =>
+    colorText &&
+    css`
+      color: ${theme.colors[colorText] || colorText};
+    `}
+  ${space}
 `;
 
 export const Tag: React.FC<TagProps> = ({
   color,
+  colorText,
   text,
+  icon,
   onClick,
   showClose,
+  ...props
 }) => {
   return (
-    <StyledTag color={color} onClick={onClick}>
-      {text || "No data"}
+    <StyledTag colorText={colorText} color={color} onClick={onClick} {...props}>
+      {icon && (
+        <div className="tag__icon">
+          <SVGIcon name={icon} size="sm" />
+        </div>
+      )}
+      <div className="tag__text">{text || "No data"}</div>
       {showClose && (
-        <span>
-          <SVGIcon name="close" color="white" size="sm" />
-        </span>
+        <div className="tag__close-button">
```

Comment:
Just FYI:
BEM notation is not needed here since the styles are automatically scoped to the component ("tag__" part is redundant).